contrib/OpenInExplorer.lua

```diff
@@ -32,82 +33,70 @@ Select the photo(s) you wish to find in explorer and press "Go to Folder".
 
 ----KNOWN ISSUES----
+Under macOS the file manager Finder opens only once and selects the file name of the last in DT selected image.
 ]]
 
 local dt = require "darktable"
 local du = require "lib/dtutils"
 local df = require "lib/dtutils.file"
 local dsys = require "lib/dtutils.system"
+local gettext = dt.gettext
 
 --Check API version
 du.check_min_api_version("5.0.0", "OpenInExplorer") 
 
-local PS = dt.configuration.running_os == "windows" and  "\\"  or  "/"
+gettext.bindtextdomain("OpenInExplorer",dt.configuration.config_dir.."/lua/locale/")
 
---Detect OS and modify accordingly--	
-local proper_install = false
-if dt.configuration.running_os ~= "macos" then
-  proper_install = true
-else
-  dt.print_error('OpenInExplorer plug-in only supports Windows and Linux at this time')
-  return
+local function _(msgid)
+    return gettext.dgettext("OpenInExplorer", msgid)
 end
 
+local act_os = dt.configuration.running_os
+local PS = act_os == "windows" and  "\\"  or  "/"
 
--- FUNCTIONS --
-
-local function open_in_explorer() --Open in Explorer
-  local images = dt.gui.selection()
-  local curr_image = ""
-  if #images == 0 then
-    dt.print('please select an image')
-  elseif #images <= 15 then
-    for _,image in pairs(images) do 
-      curr_image = image.path..PS..image.filename
-      local run_cmd = "explorer.exe /select, "..curr_image
-      dt.print_log("OpenInExplorer run_cmd = "..run_cmd)
-      resp = dsys.external_command(run_cmd)
-    end
-  else
-    dt.print('please select fewer images (max 15)')
-  end
+--Detect OS and quit if not one of macOS, Windows or Linux--	
+local proper_install = true
+if act_os ~= "macos" and act_os ~= "windows" and act_os ~= "linux" then
+  proper_install = false
+  dt.print_error(_('OpenInExplorer plug-in only supports macOS, Windows and Linux at this time'))
+  return
 end
 
-local function open_in_nautilus() --Open in Nautilus
+--The commands to open the OSes' file managers
+local fmng_cmd = {}
+fmng_cmd.linux = [[busctl --user call org.freedesktop.FileManager1 /org/freedesktop/FileManager1 org.freedesktop.FileManager1 ShowItems ass 1 ]]
+fmng_cmd.macos = 'open -R '
+fmng_cmd.windows = 'explorer.exe /select, '
+
+--The working function that opens the file manager with the image file name selected.
+local function open_in_fmanager(os, fmcmd)
   local images = dt.gui.selection()
   local curr_image = ""
+  local run_cmd = ""
   if #images == 0 then
-    dt.print('please select an image')
+    dt.print(_('Please select an image'))
   elseif #images <= 15 then
-    for _,image in pairs(images) do 
+    for _,image in pairs(images) do
       curr_image = image.path..PS..image.filename
-      local run_cmd = [[busctl --user call org.freedesktop.FileManager1 /org/freedesktop/FileManager1 org.freedesktop.FileManager1 ShowItems ass 1 ]] .. df.sanitize_filename("file://"..curr_image) .. [[ ""]]
+      if os == 'linux' then
+        run_cmd = fmcmd .. df.sanitize_filename("file://"..curr_image) .. [[ ""]]

```

Comment:
This works if you open a single file.  Repeated calls to busctl only honors the last one.  My solution would be

```
local fmng_cmd = {}
fmng_cmd.linux = [[busctl --user call org.freedesktop.FileManager1 /org/freedesktop/FileManager1 org.freedesktop.FileManager1 ShowItems ass %d %s""]]
fmng_cmd.macos = 'open -R '
fmng_cmd.windows = 'explorer.exe /select, '
local file_uris = ""

--The working function that opens the file manager with the image file name selected.
local function open_in_fmanager(os, fmcmd)
  local images = dt.gui.selection()
  local curr_image = ""
  local run_cmd = ""
  if #images == 0 then
    dt.print(_('Please select an image'))
  elseif #images <= 15 then
    for _,image in pairs(images) do
      curr_image = image.path..PS..image.filename
      if os == 'linux' then
        --run_cmd = fmcmd .. df.sanitize_filename("file://"..curr_image) .. [[ ""]]
        file_uris = file_uris .. df.sanitize_filename("file://" .. curr_image) .. " "
        dt.print_log("file_uris is " .. file_uris)
      else
        run_cmd = fmcmd .. df.sanitize_filename(curr_image)
        dt.print_log("OpenInExplorer run_cmd = "..run_cmd)
        resp = dsys.external_command(run_cmd)
      end
      -- dt.print_log("OpenInExplorer run_cmd = "..run_cmd)
      -- resp = dsys.external_command(run_cmd)
    end
    if os == 'linux' then
      run_cmd = string.format(fmng_cmd.linux, #images, file_uris)
      dt.print_log("OpenInExplorer run_cmd = "..run_cmd)
      resp = dsys.external_command(run_cmd)
    end
  else
    dt.print(_('Please select fewer images (max 15)'))
  end
end

```

Reply:
If this is ok for Windows and for Linux Nautilus, it is ok for macOS anyway. I thought this same procedure could work for Explorer and for Finder, because this would simplify the code, but it didn't work satisfactorily for Finder. So I'll leave it as it is. In the linux run_cmd line I changed "fmng_cmd.linux" for "fmcmd".

From this I conclude that the original script didn't work the way it should have on Linux. Is this right?

Reply:
Yes, the original was supposed to open a window for each selected image, but it didn't work correctly.  When I tested this version I went back to the original and tested it to compare.  When it didn't open a window with all the files selected or a window for each file selected I queried the author and he confirmed that it was intended to open a window for each. So, once you added the uri instead of the file path, you reproduced the buggy behavior of the original exactly :).  In the end, you added a feature and fixed a bug.